scripts/extract_timings.py

```diff
@@ -0,0 +1,39 @@
+#! /usr/bin/env python
+# Time-stamp: <2016-05-31 14:58:27 chrplr>
+#
+
+""" Extracts slice timing information from a dicom file.
+To be passed  as argument to slice_order for slice timing correction algorithms """
+
```

Comment:
On Mon, Jun 20, 2016 at 9:46 AM, Christophe Pallier <
notifications@github.com> wrote:

> In scripts/extract_timings.py
> <https://github.com/neurospin/pypreprocess/pull/231#discussion_r67647592>:
>
> > @@ -0,0 +1,39 @@
> > +#! /usr/bin/env python
> > +# Time-stamp: <2016-05-31 14:58:27 chrplr>
> > +#
> > +
> > +""" Extracts slice timing information from a dicom file.
> > +To be passed  as argument to slice_order for slice timing correction algorithms """
> > +
>
> well, it crashes further down the line:
>
>      Executing node 34c2260517be663ec808a9998c01a803 in dir: /home/cp983411/epi-example-dicom/essai/nifti/cache_dir/nipype_mem/nipype-interfaces-spm-preprocess-SliceTiming/34c2260517be663ec808a9998c01a803
>
> 160620-09:42:23,548 workflow INFO:
> Runtime memory and threads stats unavailable
> Hardlinked
> /home/cp983411/epi-example-dicom/nifti/StudyID-001_SeriesNumber-000009_000001ap080059.nii
> ->
> /home/cp983411/epi-example-dicom/essai/nifti/StudyID-001_SeriesNumber-000009_000001ap080059.nii
> ...
> /home/cp983411/code/pypreprocess/pypreprocess/nipype_preproc_spm_utils.py:416:
> UserWarning: subject_data.anat=None (empty); skipping coregistration!
> subject_data.anat))
> Traceback (most recent call last):
> File "/home/cp983411/code/pypreprocess/pypreprocess.py", line 23, in
> subjects = do_subjects_preproc(sys.argv[1])
> File
> "/home/cp983411/code/pypreprocess/pypreprocess/nipype_preproc_spm_utils.py",
> line 1714, in do_subjects_preproc
> subject_data, *
>
>
>
>
>
>
>
>
>
> *preproc_params) for subject_data in subjects) File
> "/usr/lib/python2.7/dist-packages/joblib/parallel.py", line 800, in call
> while self.dispatch_one_batch(iterator): File
> "/usr/lib/python2.7/dist-packages/joblib/parallel.py", line 658, in
> dispatch_one_batch self.dispatch(tasks) File
> "/usr/lib/python2.7/dist-packages/joblib/parallel.py", line 566, in
> dispatch job = ImmediateComputeBatch(batch) File
> "/usr/lib/python2.7/dist-packages/joblib/parallel.py", line 180, in __init
> self.results = batch() File
> "/usr/lib/python2.7/dist-packages/joblib/parallel.py", line 72, in call
> return [func(*args, **kwargs) for func, args, kwargs in self.items]
> File
> "/home/cp983411/code/pypreprocess/pypreprocess/nipype_preproc_spm_utils.py",
> line 1373, in do_subject_preproc
> subject_data.hardlink_output_files(final=True)
> File "/home/cp983411/code/pypreprocess/pypreprocess/subject_data.py", line
> 451, in hardlink_output_files
> setattr(self, item, linked_filename)
> UnboundLocalError: local variable 'linked_filename' referenced before
> assignment
>
OK this is an indentation bug introduced by PR #230. Ping @mrahim. The
existence of such bugs in master tells us we badly lack code-coverage. I'm
rolling a fix for this.

> But this maybe because it is an unsual use case (?). For the *.ini:
>
> #%
> disable_realign = True
> disable_coregister = True
> disable_segment = True
> disable_normalize = True
> disable_smooth = True
>
> output_dir = essai
> report = False
> plot_tsdiffana = False
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/neurospin/pypreprocess/pull/231/files/8caa896e8ea92a2478829bd93dbdbe53d287f26a#r67647592>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAms1MZQcVV47-sRFUwLs9bD0tyRfTLuks5qNkU-gaJpZM4IqgU1>
> .
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob